src/dmd/dstruct.d

```diff
@@ -607,6 +607,53 @@ extern (C++) class StructDeclaration : AggregateDeclaration
     {
         return index < numArgTypes() ? (*argTypes.arguments)[index].type : null;
     }
+
+    final bool hasNonDisabledCtor()
+    {
+        static extern (C++) class HasNonDisabledCtorVisitor : Visitor
+        {
+            bool result;
+
+            this()
+            {
+                result = false;
+            }
+
+            alias visit = Visitor.visit;
+
+            override void visit(CtorDeclaration cd)
+            {
+                if (!(cd.storage_class & STC.disable))
+                {
+                    result = true;
+                }
```

Comment:
Here and in the `OverloadSet` overload, there is a 100% overhead in vertical space because of the optional `{}`.
Personally not found of it, but if you feel strongly about keeping it, it's fine.